Concurrency/check_website_test.go

```diff
@@ -0,0 +1,54 @@
+package concurrency
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
+// mockWebsiteChecker simulates checking a website, for testing purposes.
+func mockWebsiteChecker(url string) bool {
+	if url == "http://test.com" {
+		return false
+	}
+	return true
+}
+
+// TestCheckWebsites tests the CheckWebsites function to ensure it behaves correctly.
+func TestCheckWebsites(t *testing.T) {
+	websites := []string{
+		"http://google.com",
+		"https://www.python.org",
+		"http://test.com",
+	}
+
+	want := map[string]bool{
+		"http://google.com":      true,
+		"https://www.python.org": true,
+		"http://test.com":        false,
+	}
```

Comment:
I love the idea of having a very straightforward `want` map here. However, by also defining `websites` we create a way for future changes to lead to strange results (e.g. what happens when you add a URL to `websites` but not `want`?). It might be better to stick with only defining our list in one place (probably `want`), and then constructing our URL list from that, ah lah https://stackoverflow.com/questions/21362950/getting-a-slice-of-keys-from-a-map

Reply:
I agree .It's a good idea to define our list in one place, like 'want', to avoid future inconsistencies.